src/managers/TileManager.ts

```diff
@@ -38,16 +38,72 @@ class TileManager {
                 ),
                 session.seed,
             );
-            this.list.forEach((tile: Tile) => {
-                const state: TileState = TileState.Unused; // TODO
+
+            for (let i = 0; i < this.list.length; i++) {
+                const tile: Tile = this.list[i];
                 const dto: Nullable<TileDTO> =
                     dtos.find((dto: TileDTO) => dto.id === tile.id) || null;
+
+                let state: TileState = this.identifyTiles(
+                    dto,
+                    session.turnPlayer!,
+                    i,
+                );
                 tile.apply(state, dto);
-            });
+            }
+            this.getTilesInHand(); //TODO where to save/use
             this.onSync.emit();
         });
     }
 
+    private identifyTiles(
+        dto: Nullable<TileDTO>,
+        turnPlayer: int,
+        i: int,
+    ): TileState {
+        if (dto !== null) {
+            return TileState.Placed;
+        }
+        if (
+            i <
+            turnPlayer +
+                this.getStartOfGameTilesAmount(GeneralManager.getPlayers.length)
```

Comment:
I would suggest moving the "getStartOfGameTilesAmount" call in the callstack into the method where you call "identifyTile" and passing it via a parameter into here. This way it also only gets called once instead of for every tile.